scripts/manage-datadog-secrets.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/sh
+set -eu
+
+help() {
+  cat <<EOF
+Tries to retrieve datadog credentials from the S3 object datadog-secrets.yml.
+If the object or the bucket does not exists, it will try to retrieve the
+credentials from the environment or paas-pass. It will print the variables
+to be evaluated.
+
+It can also upload the S3 object with the credentials from the environment
+or paas-pass.
+
+Usage:
+
+  AWS_ACCOUNT=dev DATADOG_PASSWORD_STORE_DIR=~/.paas-pass DEPLOY_ENV=hector
+    ./scripts/manage-datadog-secrets.sh upload
+
+  eval \$(
+    AWS_ACCOUNT=dev DATADOG_PASSWORD_STORE_DIR=~/.paas-pass DEPLOY_ENV=hector
+    ./scripts/manage-datadog-secrets.sh retrieve
+  )
+
+EOF
+}
+
+setup_env() {
+  export PASSWORD_STORE_DIR=${DATADOG_PASSWORD_STORE_DIR}
+  secrets_uri="s3://gds-paas-${DEPLOY_ENV}-state/datadog-secrets.yml"
+}
+
+get_creds_from_env_or_pass() {
+  setup_env
+  DATADOG_API_KEY="${DATADOG_API_KEY:-$(pass "datadog/${AWS_ACCOUNT}/datadog_api_key")}"
+  DATADOG_APP_KEY="${DATADOG_APP_KEY:-$(pass "datadog/${AWS_ACCOUNT}/datadog_app_key")}"
+}
+
+upload() {
+  setup_env
+  get_creds_from_env_or_pass
+  secrets_file=$(mktemp secrets.yml.XXXXXX)
+  trap 'rm  "${secrets_file}"' EXIT
+
+  cat > "${secrets_file}" << EOF
+---
+datadog_api_key: ${DATADOG_API_KEY}
+datadog_app_key: ${DATADOG_APP_KEY}
+EOF
+
+  aws s3 cp "${secrets_file}" "${secrets_uri}"
+}
+
+retrieve() {
+  setup_env
+  if aws s3 ls "${secrets_uri}" > /dev/null ; then
+    secrets_file=$(mktemp -t datadog-secrets.XXXXXX)
+    trap 'rm  "${secrets_file}"' EXIT
+
+    aws s3 cp "${secrets_uri}" "${secrets_file}"
+    DATADOG_API_KEY="${DATADOG_API_KEY:-$(awk '/datadog_api_key/ { print $2 }' "${secrets_file}")}"
+    DATADOG_APP_KEY="${DATADOG_APP_KEY:-$(awk '/datadog_app_key/ { print $2 }' "${secrets_file}")}"
```

Comment:
Can we use `val_from_yaml.rb` here like `pipelines.sh` used to? Feels safer than grepping for strings.